### DISCUSSION TITLE
About bot functionality

### DISCUSSION THREAD

**Discussion by AnthonyAspen** [mymmrac/telego#19] [Q&A] [answered]
Hello, I'm just starting with my new bot project, and I got some question about functionality of your library.

1.
Is there any way to "mark message as readen" without sending anything to a user, just do let him know that his request was readen by my bot?
2. What can you say about this library functionality compare to https://github.com/tucnak/telebot? Frankly it's very suprising for me that such a big project as yours doesn't have a lot of attention, maybe I'm wrong about it

**Comment by mymmrac** [comment, marked as answer]
Hi 👋 
Thanks for asking 🙂 

1. I recommend to use [`sendChatAction`](https://core.telegram.org/bots/api#sendchataction) with desired action like `typing` or `upload_photo`, etc. depending on what you are actually doing in background, but keep in mind that this status will be displayed at most for 5 seconds, so if you are doing something that may take longer, it's better to send an actual message explaining what is happening. Also in case if your action requires answering callback (using [`answerCallbackQuery`](https://core.telegram.org/bots/api#answercallbackquery)), you can specify `text` and optionally `show_alert` == `true` to notify user.

2. Based on functionality, Telego just takes another way of doing things, it provides full one-to-one API as Telegram describes it, with utility and helper methods. This means that it's user responsibility to decide how to use it. On the other hand, Telebot gives you a kind of event based handling (like `net/http` package) and more "high" level API. With Telego you can decide if you want to use bare Telegram method or helpers that Telego provides (also basically same event based handlers).

3. In short, they have the same functionality, but Telego can (if you want) give you more low level API, and it provides 100% of Telegram functionality.

4. The project isn't so popular, because I didn't advertise it in any ways yet, mostly it's because I am still a work in progress both with features and documentation and a lot can change.

5. If you want to share Telego and help with getting more attention, it would be great 🙂 

> Sorry for such logn responce, just wanted to cover all, because not a lot of people know about Telego 😅 .
